### PR TITLE
Get user menues

### PR DESCRIPTION
**Pull request by bbuffington1662**

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ def menu_search():
     session['establs'] = establishments
     return render_template('menu-search.html', title='Menu Search', categories=categories.keys(), cuisines=cuisines.keys(), establishments=establishments.keys())
 
-@app.route('/menu-browse', methods=['POST'])
-def menu_browse():
+@app.route('/search-menu-browse', methods=['POST'])
```
Comment (pgayee694) [inline app/routes.py]: don't change the route of the endpoint here. Instead, have it also support at GET request in which it gets the currently logged in user and queries for their saved menus that way.

File: app/templates/menu-search.html [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <body class="code">
         <h1>{{ title }}</h1>
 
-        <form action="{{ url_for('menu_browse') }}" class="form" method="POST">
+        <form action="{{ url_for('search_menu_browse') }}" class="form" method="POST">
```
Comment (pgayee694) [inline app/templates/menu-search.html]: change back to menu_browse

File: app/utils.py [hunk 3]
```diff
@@ -150,10 +150,71 @@ def list_to_string(lst):
             res += c
 
     return res
+	
+def get_user_restaurants(userId):
```
Comment (pgayee694) [inline app/utils.py:154]: needs a test

File: app/utils.py [hunk 4]
```diff
@@ -150,10 +150,71 @@ def list_to_string(lst):
...
+	
+	for userRestaurant in userRestaurants:
+		restaurant = models.Restaurant.query.filter_by(id=userRestaurant.restaurant_id).first()
+		
+		if not restaurant.name in restaurantNames:
+			restaurantNames.append(restaurant.name)
+		
+	return restaurantNames
+
+def get_restaurant_details_by_name(restaurantName, locId, start):
```
Comment (pgayee694) [inline app/utils.py]: needs a test

File: app/utils.py [hunk 5]
```diff
@@ -150,10 +150,71 @@ def list_to_string(lst):
...
+                    restaurantInfo['menu_url']))
+                except KeyError:
+                    restaurants.append(view_models.Restaurant(restaurantInfo['name'],
+                    restaurantInfo['location']['address'],
+                    '', # randomly zomato just doesn't have photos
+                    restaurantInfo['timings'],
+                    restaurantInfo['price_range'],
+                    restaurantInfo['user_rating']['aggregate_rating'],
+                    restaurantInfo['menu_url']))
+                break;
```
Comment (pgayee694) [inline app/utils.py]: don't think there should be a semi-colon here

File: app/utils.py (hunk 4 above)
Comment (pgayee694) [inline app/utils.py]: this is not performant; it takes about 3 seconds to pull back a single restaurant. Instead, this should work like get_restaurant_details(). I think you should also see if you can accomplish this same thing with get_restaurant_details as opposed to individually searching each restaurant one at a time

File: app/routes.py [hunk 6]
```diff
@@ -49,14 +49,28 @@ def menu_browse():
 
     return render_template('menu-browse.html', restaurants=restaurants, isAdd=True)
 	
+@app.route('/menu-browse', methods=['GET', 'POST'])
```
Comment (pgayee694) [inline app/routes.py]: combine this into the original menu-browse endpoint

**Review by pgayee694 — CHANGES_REQUESTED** [review]

**Review by pgayee694 — APPROVED** [review]

**Review by cheddartot — APPROVED** [review]

**Pull request merged**